### PR TITLE
Adds qudit gates to unitary API

### PR DESCRIPTION
**Pull request by dstrain115**
- Adds SwapPow and ISwapPow Gates for Qudits
- By default, this will do a swap and iswap gate for qudits.
- An exponent can be set to change this to do sqrt iswap and sqrt swap.

### CODE REVIEW

File: unitary/alpha/qudit_gates.py [hunk 1]
```diff
@@ -32,3 +32,59 @@ def _unitary_(self):
...
+
+class QuditSwapPowGate(cirq.Gate):
+    def __init__(self, dimension: int, exponent: int = 1):
+        self.dimension = dimension
+        self.exponent = exponent
+
+    def _qid_shape_(self):
+        return (self.dimension, self.dimension)
+
+    def _unitary_(self):
```
Comment (losos0) [inline unitary/alpha/qudit_gates.py:45]: The matrix is not unitary. Some columns are all zero

File: unitary/alpha/qudit_gates.py [hunk 2]
```diff
@@ -32,3 +32,59 @@ def _unitary_(self):
...
+
+class QuditISwapPowGate(cirq.Gate):
+    def __init__(self, dimension: int, exponent: int = 1):
+        self.dimension = dimension
+        self.exponent = exponent
+
+    def _qid_shape_(self):
+        return (self.dimension, self.dimension)
+
+    def _unitary_(self):
```
Comment (losos0) [inline unitary/alpha/qudit_gates.py:75]: Same problem here

File: unitary/alpha/qudit_gates.py (hunk 1 above)
Comment (dstrain115) [reply unitary/alpha/qudit_gates.py:45]: Thanks.  I was worried this matrix was not correct.  I will take another stab at it tomorrow.

Comment (dstrain115) [reply unitary/alpha/qudit_gates.py:45]: Partially fixed these gates and added a test to check unitarity for the gates.  iSwap is still not quite right though.  (and I don't have high confidence in swap yet either).  One step closer though.

File: unitary/alpha/qudit_gates.py [hunk 3]
```diff
@@ -16,7 +16,43 @@
 import cirq
 
 
+class QuditXGate(cirq.Gate):
+    """Performs a X_0a gate.
```
Comment (everthemore) [inline unitary/alpha/qudit_gates.py:20]: Would it make sense to have a general X_{ab} gate instead (where a is not always 0, I mean)?

Comment (dstrain115) [reply unitary/alpha/qudit_gates.py]: Done.

File: unitary/alpha/qudit_gates.py [hunk 4]
```diff
@@ -16,7 +16,48 @@
...
+        arr = np.zeros((self.dimension, self.dimension))
+        arr[self.source_state, self.destination_state] = 1
+        arr[self.destination_state, self.source_state] = 1
+        for i in range(self.dimension):
+            if i != self.source_state and i != self.destination_state:
+                arr[i, i] = 1
+        return arr
+
+    def _circuit_diagram_info_(self, args):
+        return f"X(0_{self.state})"
```
[the remark this answers is not in the document]
Comment (dstrain115) [reply unitary/alpha/qudit_gates.py]: Done.

**Pull request merged**